lib/IRGen/GenStruct.cpp

```diff
@@ -329,6 +331,63 @@ namespace {
     }
   };
 
+  /// A type implementation for loadable record types imported from Clang that
+  /// are non-trivially copyable (i.e., the contain ARC fields).
+  class ClangNonTrivialRecordTypeInfo final :
```

Comment:
Does SIL type lowering need to check for non-trivial Clang types too? I think today we assume all Clang types are trivial which will result in the optimizer deleting copy_value and destroy_value instructions.

Reply:
That sounds good, thanks.